### ISSUE TITLE
Write on shell

### ISSUE THREAD

**Issue by laura-rodriguez8** [TerminalStudio/dartssh2#47]
Hi !

Is there a way to write from the code to the shell and not the IDE terminal

I'm implementing the code form the shell example and it works fine:

```
final shell = await client.shell();
stdout.addStream(shell.stdout);
stderr.addStream(shell.stderr);
stdin.cast<Uint8List>().listen(shell.write);
```

But I would like to be able to write multiple entry lines from the code to the open shell

**Comment by xtyxtyx** [comment]
Yes of course! This can be achived by calling `shell.write` directly.

```dart
import 'dart:convert';

final shell = await client.shell();

final encoder = const Utf8Encoder();
shell.write(encoder.convert('ls -l\n'));
shell.write(encoder.convert('exit\n'));
```

And the result can be collected as this: 
```dart
final output = await shell.stdout
    .cast<List<int>>()
    .transform(const Utf8Decoder())
    .join();
    
print(output);
```

**Comment by laura-rodriguez8** [comment]
Thanks!!
It works, but I had to delete  `.join()` method. I'm not shure why

**Comment by xtyxtyx** [comment]
> I had to delete .join() method

The `Future` returned by `.join` completes when the stdout stream is closed, which happens when the shell is closed.

To listen for stdout and stderr before the shell closes you can try this:

```dart
final stdout = shell.stdout.cast<List<int>>().transform(const Utf8Decoder());
await for (final chunk in stdout) {
  print('chunk: "$chunk"');
}
```

**Comment by laura-rodriguez8** [comment]
Hi again!
I've notice that what I'm printing in "chunk" is both : the output of the shell but is also the input that I'm entering to the shell. Is there a way to leave out the input of the _stdout_  ?
I need it that way because I'm processing the data that I get from the output of the shell, and having the input delays that process.

`      await for (final output in shell.stdout.cast<List<int>>().transform(
          const Utf8Decoder())
      ) {

        FlightMode watch = Provider.of<FlightMode>(context, listen: false);
    if (watch.flight_mode == null || watch.flight_mode == '' ) {
      context.read<OutPut>().getOutPut(value1: output); //provider to print and process the output in another part of the code//
    shell.write(encoder.convert(
    'lat ${positionR!.latitude.toStringAsFixed(8)} lon ${positionR!.longitude.toStringAsFixed(8)}\n')); //line that I'm writing to the shell//

    print(output);`

image of "print output"

![image](https://github.com/TerminalStudio/dartssh2/assets/79954498/929a5c8d-764a-41d5-be55-0d24cb3ee481)

Thanks in advance
